wallet/src/de/schildbach/wallet/ui/more/TransactionMetadataSettingsScreen.kt

```diff
@@ -0,0 +1,311 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.ui.more
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringArrayResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.asFlow
+import androidx.lifecycle.liveData
+import androidx.work.WorkInfo
+import de.schildbach.wallet.service.work.BaseWorker
+import de.schildbach.wallet.ui.dashpay.utils.TransactionMetadataSettings
+import de.schildbach.wallet_test.R
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import org.checkerframework.checker.units.qual.s
+import org.dash.wallet.common.data.Resource
+import org.dash.wallet.common.ui.components.ButtonLarge
+import org.dash.wallet.common.ui.components.ButtonStyles
+import org.dash.wallet.common.ui.components.DashButton
+import org.dash.wallet.common.ui.components.DashCheckbox
+import org.dash.wallet.common.ui.components.DashRadioButton
+import org.dash.wallet.common.ui.components.MyTheme
+import org.dash.wallet.common.ui.components.Style
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalCoroutinesApi::class)
+@Composable
+fun TransactionMetadataSettingsScreen(
+    onBackClick: () -> Unit,
+    onInfoButtonClick: () -> Unit,
+    onSaveToNetwork: () -> Unit,
+    viewModel: TransactionMetadataSettingsPreviewViewModel
+) {
+    val backgroundColor = MyTheme.Colors.backgroundPrimary
+    val primaryTextColor = MyTheme.Colors.textPrimary
+    val secondaryTextColor = MyTheme.Colors.textSecondary
+    val scrollState = rememberScrollState()
+    val dateFormat = SimpleDateFormat.getDateInstance(SimpleDateFormat.MEDIUM)
+
+    val filterState by viewModel.filterState.collectAsState()
+    val lastSaveWorkId by viewModel.lastSaveWorkId.collectAsState()
+    val lastSaveDate by viewModel.lastSaveDate.collectAsState()
+    val futureSaveDate by viewModel.futureSaveDate.collectAsState()
+    val hasPastTransactionsToSave by viewModel.hasPastTransactionsToSave.collectAsState()
+    val publishLiveData by viewModel.publishOperationLiveData(lastSaveWorkId ?: "").asFlow().collectAsState(Resource.canceled<WorkInfo>())
```

Comment:
We should avoid using livedata in viewModels if we're gonna keep moving to Jetpack Compose. Pure flows work better for both UI systems.

Reply:
The issue here is that WorkManager supplies Worker status with LiveData.  In the backend story, I will see how I can make convert as much as possible to Flow.